### ISSUE TITLE
[halo-pr-prep] Task 4: Update training example script

### ISSUE THREAD

**Issue by jalengg** [jalengg/PyHealth#11]
## Task 4: Update training example script

**Depends on:** Task 3 (needs refactored HALO model)

### Goal

Update \`examples/halo_mimic3_training.py\` to use PyHealth standard patterns instead of custom dataset classes.

### Files to Modify

- **Modify:** \`examples/halo_mimic3_training.py\`

#### Step 1: Update training script

Replace contents of \`examples/halo_mimic3_training.py\`:

**New pattern (PyHealth standard):**
\`\`\`python
from pyhealth.datasets import MIMIC3Dataset, split_by_patient
from pyhealth.tasks import halo_generation_mimic3_fn
from pyhealth.models.generators.halo import HALO

# STEP 1: Load MIMIC-III dataset
base_dataset = MIMIC3Dataset(
    root=\"/srv/local/data/physionet.org/files/mimiciii/1.4\",
    tables=[\"DIAGNOSES_ICD\"],
    code_mapping={\"ICD9CM\": \"CCSCM\"},
    dev=False,
    refresh_cache=False,
)

# STEP 2: Set task
sample_dataset = base_dataset.set_task(halo_generation_mimic3_fn)

# STEP 3: Split dataset
train_dataset, val_dataset, test_dataset = split_by_patient(
    sample_dataset, [0.8, 0.1, 0.1]
)

# STEP 4: Initialize HALO model
model = HALO(
    dataset=sample_dataset,
    embed_dim=256,
    n_heads=4,
    n_layers=2,
    n_ctx=128,
    dropout=0.1,
    lr=5e-4,
    batch_size=32,
    epochs=50,
    pos_loss_weight=10,
    save_dir=\"./checkpoints/halo/\",
)

# STEP 5: Train with custom training loop
model.train_model(
    train_dataset=train_dataset,
    val_dataset=val_dataset,
)
\`\`\`

**Remove:**
- \`from pyhealth.datasets import HALO_MIMIC3Dataset\` (deprecated)
- Custom dataset initialization
- Manual data loading

#### Step 2: Test script (dry run)

\`\`\`bash
python examples/halo_mimic3_training.py --help
\`\`\`
Expected: No import errors (may need MIMIC-III data to fully execute)

#### Step 3: Commit

\`\`\`bash
git add examples/halo_mimic3_training.py
git commit -m \"refactor: update HALO training example to use PyHealth patterns\"
\`\`\`

### Success Criteria

- [ ] Uses \`MIMIC3Dataset\` instead of custom dataset
- [ ] Uses \`halo_generation_mimic3_fn\` task function
- [ ] Uses \`split_by_patient\` for train/val/test split
- [ ] Uses refactored HALO model with \`SampleDataset\`
- [ ] Calls \`model.train_model()\` custom training loop
- [ ] No import errors
- [ ] Script follows PyHealth standard example patterns

### Related

- Plan: \`docs/plans/2026-02-16-halo-pyhealth-integration.md\` (lines 878-965)
- Depends on: Task 3 (refactored HALO model)
- Label: **halo-pr-prep**

**Comment by jalengg** [comment]
Task 4 complete: updated examples/halo_mimic3_training.py to use PyHealth 2.0 API. The old script used argparse, HALO_MIMIC3Dataset, HALOConfig, and the now-removed train_on_init flag. The new script follows the standard PyHealth pattern: MIMIC3Dataset -> set_task(halo_generation_mimic3_fn) -> split_by_patient -> HALO(dataset, ...) -> model.train_model(train_data, val_data). Root path placeholder is /path/to/mimic3. Commit: 5fb8ce03.

**Comment by jalengg** [comment]
Closing — Task 4 is complete and verified. The updated training example passed code review (APPROVED, no issues) and was verified clean in the T9 final check. Commit: 5fb8ce03.
